src/officehours_api/backends/zoom.py

```diff
@@ -195,38 +196,29 @@ def auth_callback(cls, request):
         request.user.profile.backend_metadata['zoom'] = zoom_meta
         request.user.profile.save()
         logger.debug("Updated Zoom backend_metadata for %s", request.user.username)
-        return redirect('/')
+        state = request.GET.get('state')
+        return redirect(state)
 
     @classmethod
-    def get_auth_url(cls, redirect_uri: str):
+    def get_auth_url(cls, redirect_uri: str, state: str):
         return (
             f"{Backend.base_url}/oauth/authorize"
             f"?response_type=code"
             f"&client_id={cls.client_id}"
             f"&scope=meeting:read%20meeting:write"
+            f"&state={state}"
             f"&redirect_uri={redirect_uri}"
         )
 
     @classmethod
-    def get_public_data(self) -> BackendDict:
+    def get_public_data(cls) -> BackendDict:
         return {
-            'name': self.name,
-            'friendly_name': self.friendly_name,
-            'docs_url': self.docs_url,
-            'telephone_num': self.telephone_num
+            'name': cls.name,
+            'friendly_name': cls.friendly_name,
+            'docs_url': cls.docs_url,
+            'telephone_num': cls.telephone_num
         }
```

Comment:
Nitpick: Do you prefer `cls`? I don't mind you making this consistent, but you didn't change the other `get_public_data` methods, and I'm seeing `self` and `cls` throughout these changes.

Reply:
Nice catch, I updated those to make them all consistent. I don't feel strongly about it, but I originally went with the `cls` on `@classmethod` convention because it clarifies that you're not getting an instance.

Reply:
TBH, I also missed some nuance here, I'm still learning about `@classmethod`. Your approach is sound.